framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -721,12 +731,44 @@ protected void addInformationFromPreconditions(S info,
 
         S info = in.getRegularStore();
         V rhsValue = in.getValueOfSubNode(rhs);
+
+        Receiver expr = FlowExpressions.internalReprOf(analysis.getTypeFactory(),
+                n.getTarget());
+
+        if (inferSignatures && !expr.containsUnknown()
+                && expr instanceof FieldAccess) {
+            // Updates .jaif file
+            FieldAccessNode lhsNode = ((FieldAccessNode)lhs);
+            ClassSymbol clazzSymbol = SignatureInferenceScenes.getEnclosingClassSymbol(
```

Comment:
This is the only call to getEnclosingClassSymbol.  Could it be called by updateFieldTypeInScene instead?  That would make SignatureInferenceScenes easier for a client to use, at least here.